### PR TITLE
v 3.0.0 Development

### LINKED ISSUE #13: respect current UserInterfaceStyle ( dark and light mode )
simply change colors like .black or .white
to uicolors with two appearances (.systemBackground, .label)

### PR DESCRIPTION
**Pull request by yervandsar**
1. Swift Package manager support
2. Performance improvements
3. New Structured file system and more clean code
4. Dark/Light mode dynamic support
5. New Protocol Oriented Options
6. Wild size support in expanded container
7. New Animations for expanded container and ability add own animation
8. New Indicator types

### CODE REVIEW

**Review by hamtiko — COMMENTED** [review]
I think you should remove `ExpandedTabBar.xcodeproj/xcuserdata/` from git and add it to gitignore, so you will avoid to have personalized configurations in git repo.

File: Sources/ExpandedTabBar/ExpandedTabBarViews.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+//  Created by Yervand Saribekyan on 3/13/18.
+//  Copyright © 2018 Yervand Saribekyan. All rights reserved.
+//
+#if !os(macOS)
+
+import UIKit
+
+internal final class ExpandedTabBarViews {
+
+    class func moreItemView(for item: UITabBarItem,
```
Comment (hamtiko) [inline Sources/ExpandedTabBar/ExpandedTabBarViews.swift]: Why to not use `static` here as you can't override it due to `internal` and `final` modifiers.

File: Sources/ExpandedTabBar/Styles/BackgroundOptions.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    
+    /// Background alpha.
+    var alpha: CGFloat { get set }
+    
+    /// Close on tap to background.
+    var closeOnTap: Bool { get set }
+}
+
+/// BackgroundOptions declaration
+public final class BackgroundDefaultOptions: BackgroundOptions {
```
Comment (hamtiko) [inline Sources/ExpandedTabBar/Styles/BackgroundOptions.swift]: Why to not use struct for this?

File: Sources/ExpandedTabBar/ExpandedTabBarViews.swift (hunk 1 above)
Comment (yervandsar) [reply Sources/ExpandedTabBar/ExpandedTabBarViews.swift]: Good point)

**Review by hamtiko — COMMENTED** [review]
Please also remove following files from git tracking:
```
project.xcworkspace/xcuserdata/yervandsar.xcuserdatad/UserInterfaceState.xcuserstate
...xcodeproj/xcuserdata/yervandsar.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist
```

File: README.md [hunk 3]
```diff
@@ -1,111 +1,209 @@
...
+moreSelectedIcon : UIImage? // Default nil
+```
+- #### *Dark Mode Support*
+**ExpandedTabBar** is fully ***Light/Dark*** mode supported and for setting any `UIColor` or `CGColor` you can use.
 ```swift
-import UIKit
-import ExpandedTabBar
+let color: UIColor/CGColor = .pattern(light: UIColor, dark: UIColor)
+```
+**NOTE:** If device OS version not supports dark mode, it will take light color by default.
```
Comment (hovaks) [inline README.md]: If device OS version does not support dark mode, it will take light as default.

**Review by hovaks — APPROVED** [review]

**Review by hamtiko — APPROVED** [review]

**Pull request merged**